lib/test/cdo/contact_rollups/test_pardot_v2.rb

```diff
@@ -265,4 +298,13 @@ def test_calculate_data_delta
       assert_equal test[:expected_delta], delta, "Test index #{index} failed"
     end
   end
+
+  # @param str a heredoc string
+  # @return Nokogiri::XML::Document
+  def create_xml_from_heredoc(str)
```

Comment:
Could you clarify why you're doing this instead of using Nokogiri (for my understanding, probably doesn't need to go in code?)

Reply:
Also -- looked up the regex you're matching here:
https://regexr.com/
Seems like it will only match whitespaces at the beginning of the first line unless you add a multiline flag `m` at the end of your regex pattern -- intentional?
https://javascript.info/regexp-multiline-mode

Reply:
This regex pattern is used in a `gsub` (global substitution) so it can processes multiple lines. The `\m` modifier is mostly useful when used with `.` character ([link](https://ruby-doc.org/core-2.5.0/Regexp.html#class-Regexp-label-Character+Classes)), which we don't need in this case.

I add another replacement to remove all whitespaces at the end of each line.